src/components/v5/common/ActionSidebar/partials/forms/SimplePaymentForm/utils.tsx

```diff
@@ -77,3 +82,82 @@ export const simplePaymentDescriptionMetadataGetter: DescriptionMetadataGetter<
     },
   );
 };
+
+type PaymentPayload = OneTxPaymentPayload['payments'][number];
+
+interface GetPaymentPayloadParams {
+  colony: Colony;
+  recipientAddress: string;
+  amount: string;
+  tokenAddress: string;
+  networkInverseFee?: string;
+}
+
+const getPaymentPayload = ({
+  colony,
+  recipientAddress,
+  amount,
+  tokenAddress,
+  networkInverseFee,
+}: GetPaymentPayloadParams): PaymentPayload => {
+  const selectedToken = getSelectedToken(colony, tokenAddress);
+  const decimals = getTokenDecimalsWithFallback(selectedToken?.decimals);
+
+  const amountWithFees = networkInverseFee
+    ? calculateFee(amount, networkInverseFee, decimals).totalToPay
+    : amount;
+
+  return {
+    recipient: recipientAddress,
+    tokenAddress,
+    amount: amountWithFees, // @NOTE: Only the contract sees this amount
+    decimals,
+  };
+};
+
+export const getSimplePaymentPayload = (
+  colony: Colony,
+  values: SimplePaymentFormValues,
+  networkInverseFee: string | undefined,
+) => {
+  const {
+    from,
+    description,
+    createdIn,
+    title,
+    amount,
+    recipient: recipientAddress,
+    payments,
+  } = values;
+  const fromDomainId = Number(from);
+  const createdInDomainId = Number(createdIn);
+
+  const transformedPayload: OneTxPaymentPayload = {
+    colonyName: colony.name,
+    colonyAddress: colony.colonyAddress,
+    domainId: fromDomainId,
+    payments: [
+      getPaymentPayload({
+        colony,
+        recipientAddress,
+        amount: amount.amount.toString(),
+        tokenAddress: amount.tokenAddress,
+        networkInverseFee,
+      }),
+      ...payments.map(({ recipient, amount: paymentAmount }) =>
+        getPaymentPayload({
+          colony,
+          recipientAddress: recipient,
+          amount: paymentAmount.amount.toString(),
+          tokenAddress: amount.tokenAddress,
```

Comment:
you know best, I'm dropping a comment just in case of a typo, do we need `paymentAmount` instead of just `amount` here?  :)

Reply:
Yes, `amount` is a property of the `values` argument, whereas `paymentAmount` is a property of one of the `payments` array elements